operators/pkg/controller/common/defaults/pod_template.go

```diff
@@ -287,3 +287,14 @@ func (b *PodTemplateBuilder) WithInitContainers(initContainers ...corev1.Contain
 
 	return b
 }
+
+// WithResources sets up the given default resource requirements if both resources limits and requests
```

Comment:
godoc not sync: WithDefaultResources*

Reply:
No strong opinion but I think I would have call it `WithResources(resources corev1.ResourceRequirements)`.

Reply:
You're right, it better matches other functions of this file: apply our values, unless already set by the user.